src/proxy/inbound.rs

```diff
@@ -363,7 +363,33 @@ impl Inbound {
             // Translate service port to target port
             match svc.ports.get(&hbone_addr.port()) {
                 Some(port) => {
-                    dest_port = *port;
+                    // Support named target ports
+                    if *port == 0 {
+                        // Get endpoints from the destination workload uid
+                        let endpoint =
+                            svc.endpoints
+                                .get(&destination_workload.uid)
+                                .ok_or_else(|| {
+                                    InboundError(
+                                        Error::NoWorkloadEndpoints(
+                                            destination_workload.to_string(),
+                                        ),
+                                        StatusCode::BAD_REQUEST,
+                                    )
+                                })?;
+                        // Get the target port from the endpoint
+                        dest_port = *endpoint.port.get(&hbone_addr.port()).ok_or_else(|| {
```

Comment:
lmk if there is a better way to do this in Rust. Looking into it - just looks stinky